BT_DEBUG_MONITOR_UART missing dependency for nRF5340?

**Describe the bug**
If you enable `CONFIG_BT_DEBUG_MONITOR_UART` and disable `CONFIG_UART_CONSOLE` as per https://docs.zephyrproject.org/latest/connectivity/bluetooth/bluetooth-dev.html#embedded-hci-tracing without enabling `CONFIG_BT_DEBUG_LOG` then the following linker error occurs:

```
/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/subsys/bluetooth/host/libsubsys__bluetooth__host.a(monitor.c.obj): in function `monitor_log_process':
dev/nordic/zephyrproject/zephyr/subsys/bluetooth/host/monitor.c:325: undefined reference to `log_output_msg_process'
```

If you remove `CONFIG_UART_CONSOLE=n` or if you add `CONFIG_BT_DEBUG_LOG=y` then the linker error is resolved, which indicates some missing dependency somewhere. 

**To Reproduce**
Steps to reproduce the behavior:
1. Add `CONFIG_BT_DEBUG_MONITOR_UART=y` and `CONFIG_UART_CONSOLE=n` to https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/bluetooth/peripheral/prj.conf
2. Build https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/bluetooth/peripheral with `-b nrf5340dk_nrf5340_cpuapp` 
3. See error

If you remove `CONFIG_UART_CONSOLE=n` or add `CONFIG_BT_DEBUG_LOG=y` then there are no build errors. Similarly if building for `nrf52840dk_nrf52840` instead of `nrf5340dk_nrf5340_cpuapp`, the issue also does not appear. 

**Expected behavior**
Expect to be able to follow Zephyr documentation and build a sample with support for HCI logs in `btmon`. 

**Impact**
Minor

**Logs and console output**
N/A

**Environment (please complete the following information):**
 - Commit SHA or Version used: 39433f0669f1fd620f36605d1c3df61a6cf97ae2

**Additional context**
N/A

--- a/subsys/bluetooth/common/Kconfig
+++ b/subsys/bluetooth/common/Kconfig
@@ -282,6 +282,7 @@ endif # BT_ASSERT
 
 config BT_MONITOR
 	bool
+	select LOG_OUTPUT
 
 choice BT_DEBUG_TYPE
 	prompt "Bluetooth debug type"

